packages/rocketchat-livechat/.app/client/lib/LivechatFileUpload.js

```diff
@@ -0,0 +1,187 @@
+/* globals, fileUpload, swal, Livechat, Handlebars */
+/* exported LivechatFileUpload */
+import visitor from '../../imports/client/visitor';
+import s from 'underscore.string';
+import request from 'request';
+
+function sendFileRequest(file, roomId, visitorToken) {
+	const url = `${ Meteor.absoluteUrl() }api/v1/livechat/upload/${ roomId }`;
+	const form = new FormData();
+	form.append('file', file)
+	
+	const request = new XMLHttpRequest();
+	request.open("POST", url);
+	request.setRequestHeader("X-Visitor-Token", visitorToken);
+
+	request.onload = function () {
+		if (request.status !== 200) {	
+			showError(request.statusText);					
+		}
+	};
+	/*
+	request.onprogress = function () {
+		console.log('LOADING', request.status);
+	};
+	*/
+	request.onerror = function () {
+		showError(request.statusText);
+	};
+
+	request.send(form);		  
+}
+
+function readAsDataURL(file, callback) {
+	const reader = new FileReader();
+	reader.onload = ev => callback(ev.target.result, file);
+
+	return reader.readAsDataURL(file);
+}
+
+function getUploadPreview(file, callback) {
+	// If greater then 10MB don't try and show a preview
+	if (file.file.size > (10 * 1000000)) {
+		return callback(file, null);
+	} else if (file.file.type == null) {
+		callback(file, null);
+	} else if ((file.file.type.indexOf('audio') > -1) || (file.file.type.indexOf('video') > -1) || (file.file.type.indexOf('image') > -1)) {
+		file.type = file.file.type.split('/')[0];
+		return readAsDataURL(file.file, content => callback(file, content));
+	} else {
+		return callback(file, null);
+	}
+}
+
+function formatBytes(bytes, decimals) {
+	if (bytes === 0) {
+		return '0 Bytes';
+	}
+
+	const k = 1000;
+	const dm = (decimals + 1) || 3;
+
+	const sizes = [
+		'Bytes',
+		'KB',
+		'MB',
+		'GB',
+		'TB',
+		'PB'
+	];
+
+	const i = Math.floor(Math.log(bytes) / Math.log(k));
+
+	return `${ parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) } ${ sizes[i] }`;
+}
+
+function sendFileMessage(file, roomId) {
+	if (!visitor.isSubscribed(roomId)) {	
+		Meteor.call('livechat:startFileUploadRoom', roomId, visitor.getToken(), (error, result) => {
+			if (error) {
+				return showError(error.message);
+			}
+
+			visitor.subscribeToRoom(result.room._id);
+			visitor.setRoom(result.room._id);
+			sendFileRequest(file, roomId, visitor.getToken());
+			parentCall('callback', 'chat-started');
+		});
+	} else {
+		sendFileRequest(file, roomId, visitor.getToken());
+	}
+}
+
+fileUpload = function(file) {
+	Meteor.call('livechat:checkTypeFileUpload', file.file.type, (error, result) => {
```

Comment:
IMO both checks (type and size) should be done by the same method. so instead of calling multiple methods for multiples validations, I would rather then call a single `livechat:validateFileUpload` and that method would than perform all validations needed.